src/pages/openSourceApply.js

```diff
@@ -0,0 +1,170 @@
+    };
+  }
+
+  static propTypes = {
+    token: PropTypes.string,
+    loadingLoggedInUser: PropTypes.bool,
+    LoggedInUser: PropTypes.object,
+    createCollectiveFromGithub: PropTypes.func,
+  };
+
+  state = {
+    result: {},
+    loadingRepos: false,
+    repositories: [],
+    creatingCollective: false,
+  };
+
+  async componentDidMount() {
+    const { token } = this.props;
+    if (!token) {
+      return;
+    }
+    this.setState({ loadingRepos: true });
+
+    try {
+      const repositories = await getGithubRepos(token);
+      if (repositories.length !== 0) {
+        this.setState({ repositories, loadingRepos: false, result: {} });
+      } else {
+        this.setState({
+          loadingRepos: false,
+          result: { type: 'info', mesg: 'Info: No Repository found' },
+        });
+      }
+    } catch (error) {
+      this.setState({
+        loadingRepos: false,
+        result: { type: 'error', mesg: 'Error: An unknown error occured' },
+      });
+    }
+  }
+
+  renderContent() {
+    const { token, LoggedInUser } = this.props;
+    const { repositories } = this.state;
+
+    if (!LoggedInUser) {
+      return <SignInOrJoinFree redirect={Router.asPath} />;
+    } else if (!token || repositories.length === 0) {
+      return this.renderConnectGithubButton();
+    } else {
+      return this.renderGithubRepos();
+    }
+  }
+
+  async createCollectives(collectiveInputType) {
+    collectiveInputType.type = 'COLLECTIVE';
+    try {
+      const res = await this.props.createCollectiveFromGithub(collectiveInputType);
+      const collective = res.data.createCollectiveFromGithub;
+      Router.pushRoute('collective', {
+        slug: collective.slug,
+        status: 'collectiveCreated',
+        CollectiveId: collective.id,
+        CollectiveSlug: collective.slug,
+      });
+    } catch (err) {
+      console.error('>>> createCollective error: ', JSON.stringify(err)); // TODO - Remove
+      const errorMsg = err.graphQLErrors && err.graphQLErrors[0] ? err.graphQLErrors[0].message : err.message;
+      this.setState({
+        creatingCollective: false,
+        result: { type: 'error', mesg: errorMsg },
+      });
+      // throw new Error(errorMsg);
+    }
+  }
+
+  renderGithubRepos() {
+    const { repositories, creatingCollective } = this.state;
+    if (repositories.length !== 0) {
+      return (
+        <StyledInputField htmlFor="collective">
+          {fieldProps => (
+            <GithubRepositories
+              {...fieldProps}
+              repositories={repositories}
+              creatingCollective={creatingCollective}
+              onCreateCollective={data => {
+                this.setState({ creatingCollective: true });
+                this.createCollectives(data);
+              }}
+            />
+          )}
+        </StyledInputField>
+      );
+    }
+  }
+
+  renderConnectGithubButton() {
+    const connectUrl = `/api/connected-accounts/github?redirect=${WEBSITE_URL}/opensource/apply`;
```

Comment:
Instead of WEBSITE_URL, can you use `getBrowserWebsiteUrl` which was introduced recently.

On feature branches pushed on the `now` platform, `WEBSITE_URL` is not properly set.

Reply:
Noted

Reply:
@flickz Can you integrate that feedback?